core/include/scipp/core/dataset.h

```diff
@@ -494,12 +494,20 @@ template <class Base> class MutableProxy : public Base {
 
 /// Const proxy for Dataset, implementing slicing and item selection.
 class SCIPP_CORE_EXPORT DatasetConstProxy {
+  explicit DatasetConstProxy() : m_dataset(nullptr) {}
+
 public:
   explicit DatasetConstProxy(const Dataset &dataset) : m_dataset(&dataset) {
     for (const auto &item : dataset.m_data)
       m_indices.emplace_back(item.first);
   }
 
+  static DatasetConstProxy makeShallowProxy(const Dataset &dataset) {
```

Comment:
The name is very unclear. How about making a more generic constructor that accepts a list of data items to keep:

```cpp
explicit DatasetConstProxy(const Dataset &dataset, const std::vector<std::string> &keys) : m_dataset(&dataset) {
    for (const auto &key : keys)
      m_indices.emplace_back(...);
  }
```

Then use it with an empty list of keys to get what `makeShallowProxy` does?